lib/hydra/works/models/concerns/generic_file/characterization.rb

```diff
@@ -0,0 +1,116 @@
+      class_attribute :parser_config
+      self.parser_config = { exif_tool_version: :exif_version,
+                             file_mime_type: :mime_type }
+
+      class_attribute :parser_class
+      self.parser_class = FitsDatastream
+
+      # Extract the metadata from a content file and store it as properties.
+      # The target properties must already exist.
+      # @param [ActiveFedora::File] file to characterize
+      def characterize(file = original_file)
+        # Run fits and get xml string
+        extracted_md = extract_metadata(file)
+        # Parse xml string to get fits datastream terms and values
+        terms = parse_metadata(extracted_md)
+        # Store terms for which there are properties mapped.
+        store_metadata(terms)
+      end
+
+      protected
+
+      def extract_metadata(file = original_file)
+        return unless file.has_content?
+        Hydra::FileCharacterization.characterize(file.content, file.original_name, :fits) do |cfg|
+          cfg[:fits] = Hydra::Derivatives.fits_path
+        end
+      end
+
+      # Use OM to parse metadata
+      def parse_metadata(metadata)
+        datastream = parser_class.new
+        datastream.ng_xml = metadata if metadata.present?
+        characterization_terms(datastream)
+      end
+
+      # Get terms and values
+      # This is using the names of the proxy terms; not the name of what the proxy points to.
+      def characterization_terms(datastream)
+        h = {}
+        datastream.class.terminology.terms.each_pair do |key, target|
+          # a key is a proxy if its target responds to proxied_term
+          next unless target.respond_to? :proxied_term
+          begin
+            h[key] = datastream.send(key)
+          rescue NoMethodError
+            next
+          end
+        end
+        h.delete_if { |_k, v| v.empty? }
+      end
+
+      # Assign values of the instance properties from the metadata mapping :prop => val
+      def store_metadata(terms)
+        # Go through each term.
+        terms.each_pair do |term, value|
+          # Figure out property to use or nil
+          property = property_for(term)
+          next if property.nil?
+          # Array-ify the value to avoid a conditional here
+          Array(value).each { |v| append_property_value(property, v) }
+        end
+      end
+
+      # Check mapping then self for matching term.
+      # Return property symbol or nil
+      def property_for(term)
+        if parser_config.key?(term) && self.respond_to?(parser_config[term])
+          parser_config[term]
+        else
+          term if self.respond_to?(term)
+        end
+      end
+
+      def append_property_value(property, value)
+        property_value = send(property)
+        if property_value.is_a? Array
+          property_value << value
+        else
+          send("#{property}=", value)
+        end
+      end
+    end # of included do
```

Comment:
The comment or the two short functions above?  The comment was to keep my place when the module had a mix of class and instance methods.  The functions were my response to rubocop calling out functions that were too verbose or too complex.

Comment removed.  Not willing to shoot it out with rubocop about the functions again.